File: 11505 - Logo.java
/*
*	User: Isanchez_Aguilar
*	Problem: UVA 11505 - Logo
*/
import java.io.OutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.PrintWriter;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		InputStream inputStream = System.in;
		OutputStream outputStream = System.out;
		Scanner in = new Scanner(inputStream);
		PrintWriter out = new PrintWriter(outputStream);
		UVA11505 solver = new UVA11505();
		int testCount = Integer.parseInt(in.next());
		for (int i = 1; i <= testCount; i++)
			solver.solve(i, in, out);
		out.close();
	}

	static class UVA11505 {
		public void solve(int testNumber, Scanner in, PrintWriter out) {
			double dx = 0.0;
			double dy = 0.0;
			double angle = 0.0;
			int n = in.nextInt();

			for (int i = 0; i < n; ++i) {
				String command = in.next();
				double dist = in.nextDouble();

				switch (command.charAt(0)) {
					case 'l':
						angle += dist;
						angle %= 360.0;
						break;
					case 'r':
						angle -= dist;
						angle %= 360.0;
						break;
					case 'f':
						dx += (dist * Math.cos(Math.toRadians(angle)));
						dy += (dist * Math.sin(Math.toRadians(angle)));
						break;
					default:
						dx -= (dist * Math.cos(Math.toRadians(angle)));
						dy -= (dist * Math.sin(Math.toRadians(angle)));
				}
			}

			out.println(Math.round(Math.hypot(dx, dy)));
		}
	}
}

File: 11505 - Logo.cpp
/*
*	User: Isanchez_Aguilar
*	Problem: UVA 11505 - Logo
*/
#include <bits/stdc++.h>

using namespace std;

typedef double Double;

const Double PI = M_PI;
const Double EPS = 1e-2;

inline Double isZero(Double a) { return fabs(a) < EPS; }

inline Double toRadians(Double angle) { return PI * angle / 180.0; }

class Vector
{
public:
	Double x;
	Double y;

	Vector():x(0), y(0) {}

	Vector(Double _x, Double _y):x(_x), y(_y) {}

	inline void move(Double hypotenuse, Double angle)
	{	
		x += hypotenuse * cos(toRadians(angle));
		y += hypotenuse * sin(toRadians(angle));
	}

	inline Double norm() const { return hypot(x, y); }
};

int main(void)
{
	ios_base::sync_with_stdio(0);
	cin.tie(0);

	int testCases;
	cin >> testCases;

	while (testCases--)
	{
		int instructions;
		cin >> instructions;

		Double angle = 0.0;
		Vector position(0.0, 0.0);

		while (instructions--)
		{
			Double value;
			string instruction;
			cin >> instruction >> value;

			switch (instruction[0])
			{
				case 'l':
					angle = fmod(angle + value, 360.0);
				break;
				
				case 'r':
					angle = fmod(angle - value, 360.0);
				break;

				case 'f':
					position.move(value, angle);
				break;
				
				default:
					position.move(-value, angle);
			}
		}

		cout << (int)round(position.norm()) << "\n";
	}

	return 0;
}